addons/binding/org.openhab.binding.geofence/src/main/java/org/openhab/binding/geofence/internal/BindingComponent.java

```diff
@@ -0,0 +1,264 @@
+                    if (configAdminRef != null) {
+                        logger.debug("ConfigurationAdmin service was found.");
+                        ConfigurationAdmin configAdmin = bundleContext.getService(configAdminRef);
+                        org.osgi.service.cm.Configuration configuration = configAdmin.getConfiguration(CONFIG_PID);
+                        Dictionary<String, Object> properties = copy(configuration.getProperties());
+
+                        if (properties != null) {
+                            logger.debug("Bundle configuration was found.");
+                            PointType location = locationProv.getLocation();
+                            String locationConfig = (String) properties.get(BindingConstants.CONFIG_LOCATION);
+                            if (location != null) {
+                                if (locationConfig == null || !location.toFullString().equals(locationConfig)) {
+                                    logger.debug("Updating location from system config: {}", location.toFullString());
+                                    properties.put(BindingConstants.CONFIG_LOCATION, location.toFullString());
+                                    configuration.update(properties);
+                                }
+                            } else if (locationConfig != null) {
+                                logger.debug("Clear primary location config as system location is missing");
+                                properties.remove(BindingConstants.CONFIG_NAME);
+                                properties.remove(BindingConstants.CONFIG_LOCATION);
+                                properties.put(BindingConstants.CONFIG_RADIUS, 100);
+                                configuration.update(properties);
+                            }
+                        }
+                    }
+                } catch (IOException e) {
+                    logger.error("Unable to update primary location from system config.", e);
+                }
+            }
+        }
+    }
+
+    private Dictionary<String, Object> copy(Dictionary<String, Object> properties) {
+        if (properties != null) {
+            Hashtable<String, Object> newProp = new Hashtable<>();
+
+            Enumeration<String> keys = properties.keys();
+            while (keys.hasMoreElements()) {
+                String key = keys.nextElement();
+                newProp.put(key, properties.get(key));
+            }
+
+            return newProp;
+        }
+        return null;
+    }
+
+    /**
+     * Deactivate the binding. It stops the HTTP callback endpoint and stops the device discovery service.
+     *
+     * @param componentContext Component context.
+     */
+    protected void deactivate(ComponentContext componentContext) {
+        logger.info("Deactivating Geofence Binding");
+        logger.debug("Stopping callback servlets");
+        otHTTPEndpoint.deactivate();
```

Comment:
remove the deactivate method and simply unregister the servlets from the httpService here.